core/src/main/java/org/elasticsearch/env/NodeEnvironment.java

```diff
@@ -1003,8 +1003,12 @@ private static void tryWriteTempFile(Path path) throws IOException {
         if (Files.exists(path)) {
             Path resolve = path.resolve(".es_temp_file");
             try {
-                Files.createFile(resolve);
-                Files.deleteIfExists(resolve);
```

Comment:
And anyway, moving it to a `finally` block does help because if the create fails because the file already exists, the delete in the `finally` block will clean it up so we only fail in this dumb way once.